File: backend/VocabuAI.Api/Dtos/PronunciationLookupResponseDto.cs
namespace VocabuAI.Api.Dtos;

// Keep in sync with frontend/src/domain/dtos/PronunciationLookupResponseDto.ts

public sealed record PronunciationLookupResponseDto(
    string Term,
    string LanguageCode,
    bool IsAvailable,
    string? AudioUrl,
    string? AttributionUrl,
    string? LicenseShortName,
    string? Creator,
    string? Credit,
    string? Source,
    string? Message
);

File: frontend/src/domain/dtos/PronunciationLookupResponseDto.ts
// Keep in sync with backend/VocabuAI.Api/Dtos/PronunciationLookupResponseDto.cs
export type PronunciationLookupResponseDto = {
  term: string;
  languageCode: string;
  isAvailable: boolean;
  audioUrl: string | null;
  attributionUrl: string | null;
  licenseShortName: string | null;
  creator: string | null;
  credit: string | null;
  source: string | null;
  message: string | null;
};
